com.unity.render-pipelines.universal/Runtime/2D/Renderer2DData.cs

```diff
@@ -190,5 +190,41 @@ internal override Shader GetDefaultShader()
             return Shader.Find("Universal Render Pipeline/2D/Sprite-Lit-Default");
         }
 #endif
+
+
+        internal void InitializeTransient()
+        {
+            for (var i = 0; i < m_LightBlendStyles.Length; ++i)
+            {
+                if(m_LightBlendStyles[i].renderTargetHandle.id == 0)
+                    m_LightBlendStyles[i].renderTargetHandle.Init($"_ShapeLightTexture{i}");
+            }
+
+            if(lightMaterials == null || lightMaterials.Length == 0)
+                lightMaterials= new Material[k_NumberOfLightMaterials];
+
+            if (normalsRenderTarget.id == 0)
+                normalsRenderTarget.Init("_NormalMap");
+
+            if (shadowsRenderTarget.id == 0)
+                shadowsRenderTarget.Init("_ShadowTex");
+
+            const int totalMaterials = 256;
+
+            if(shadowMaterials == null || shadowMaterials.Length == 0)
+                shadowMaterials = new Material[totalMaterials];
+            if(removeSelfShadowMaterials == null || removeSelfShadowMaterials.Length == 0)
+                removeSelfShadowMaterials = new Material[totalMaterials];
+        }
+
+        const int k_NumberOfLightMaterials = 1 << 5 + 3;  // 5 keywords +  volume bit, shape bit
```

Comment:
Can we move this line to where `k_NumberOfLightMaterials` is first used?

Reply:
i'll restore it for now, but these things should be use with hashmap